core/services/workflows/syncer/handler.go

```diff
@@ -284,28 +355,126 @@ func (h *eventHandler) workflowRegisteredEvent(
 	return nil
 }
 
-// workflowUpdatedEvent handles the WorkflowUpdatedEvent event type.
+// workflowUpdatedEvent handles the WorkflowUpdatedEvent event type by first finding the
+// current workflow engine, stopping it, and then starting a new workflow engine with the
+// updated workflow spec.
 func (h *eventHandler) workflowUpdatedEvent(
-	_ context.Context,
-	_ WorkflowRegistryEvent,
+	ctx context.Context,
+	payload WorkflowRegistryWorkflowUpdatedV1,
 ) error {
-	return ErrNotImplemented
+	oldWorkflowID := hex.EncodeToString(payload.OldWorkflowID[:])
+
+	// stop the old workflow engine
+	e, err := h.engineRegistry.Pop(oldWorkflowID)
```

Comment:
Log a warning if you can't pop rather than interrupting altogether and proceed with the rest.